app/helpers/application_helper.rb

```diff
@@ -1,2 +1,17 @@
 module ApplicationHelper
+  def current_user
+    if session[:user_id]
+      @current_user ||= User.find_by(id: session[:user_id])
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user && user.authenticated?(cookies[:remember_token])
+        sign_in user
+        @current_user = user
+      end
+    end
+  end
+
+  def signed_in?
+    !current_user.nil?
+  end
```

Comment:
There should be no application logic in the views, you already implemented those methods in the `ApplicationController`